### PR TITLE
Android support

### PR DESCRIPTION
**Pull request by 3rob3**

### CODE REVIEW

**Comment by BlakeFL** [comment]
Android! LETS GO!

**Comment by JW-CH** [comment]
image
image
@BlakeFL some progress on the android version :)

**Comment by BlakeFL** [comment]
Looks great.  I need to start making a case for an old Kindle Fire tablet to prepare for this.

File: ImmichFrame/Models/Settings.cs [hunk 1]
```diff
@@ -38,13 +43,88 @@ public static Settings CurrentSettings
...
+            {
+                using (XmlWriter writer = XmlWriter.Create(sww))
+                {
+                    xsSubmit.Serialize(writer, this);
+                    xml = sww.ToString();
+                }
+            }
+
+            // TODO: some error handling with try catch?
+            File.WriteAllText(AppDomain.CurrentDomain.BaseDirectory + @"Settings.xml", xml);
```
Comment (3rob3) [inline ImmichFrame/Models/Settings.cs:127]: I'd like to re-think this serialize section in the future. I'd prefer to keep XML and the built-in app settings separate.

Comment (JW-CH) [reply ImmichFrame/Models/Settings.cs:127]: This is still separate, or am I missing something?

Comment (3rob3) [reply ImmichFrame/Models/Settings.cs:127]: Overwriting the XML.

Comment (3rob3) [reply ImmichFrame/Models/Settings.cs:127]: My preference would be if the XML is there then use it, don't modify it, and potentially don't even allow settings screen to be shown. If XML is not there, then use app settings and allow settings screen, same as Android.

**Comment by hunkyn** [comment]
I have an old fire tablet and the apk is failing. Is the a requirement for OS version or something that should be met?

**Comment by 3rob3** [comment]
> I have an old fire tablet and the apk is failing. Is the a requirement for OS version or something that should be met?
minSdkVersion="28", which is Android 9. What version is yours?

**Comment by hunkyn** [comment]
> > I have an old fire tablet and the apk is failing. Is the a requirement for OS version or something that should be met?
> 
> minSdkVersion="28", which is Android 9. What version is yours?
> 
> 

This is fire tablet and it has the latest version. Is there a way i can see the SDK version for fireOS?

**Comment by JW-CH** [comment]
> > > I have an old fire tablet and the apk is failing. Is the a requirement for OS version or something that should be met?
> > 
> > 
> > minSdkVersion="28", which is Android 9. What version is yours?
> 
> This is fire tablet and it has the latest version. Is there a way i can see the SDK version for fireOS?
> 
> EDIT: Seems like this fire tablet does not support Android 9. Atleast fireOS.

We could try to lower the required OS version. We'll check if lower android versions could be supported.

**Pull request merged**